db/upgrade.php

```diff
@@ -91,5 +91,15 @@ function xmldb_auth_mumie_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2023062000, 'auth', 'mumie');
     }
 
+    if ($oldversion < 2023110800) {
+        $table = new xmldb_table('auth_mumie_cryptographic_key');
+        $field = new xmldb_field('key');
+        if ($dbman->field_exists($table, $field)) {
+            $field->set_attributes(XMLDB_TYPE_TEXT, null, null, XMLDB_NOTNULL, null, null);
```

Comment:
what is the reason for setting these attributes?

Reply:
Otherwise its not working. This is a snippet from the moodle forum.

Reply:
so did the attributes change?

Reply:
No, just the field name :)